nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/ftp/TestVirtualFileSystemView.java

```diff
@@ -0,0 +1,223 @@
+
+        // THEN
+        assertHomeDirectory("/");
+        assertCurrentDirectory("/");
+    }
+
+    @Test
+    public void testChangeToSameDirectory() throws FtpException {
+
+        // WHEN
+        fileSystemView.changeWorkingDirectory(".");
+
+        // THEN
+        assertHomeDirectory("/");
+        assertCurrentDirectory("/");
+    }
+
+    @Test
+    public void testChangeToSameDirectoryNonRoot() throws FtpException {
+        // GIVEN
+        fileSystemView.changeWorkingDirectory("/Directory1");
+
+        // WHEN
+        fileSystemView.changeWorkingDirectory(".");
+
+        // THEN
+        assertHomeDirectory("/");
+        assertCurrentDirectory("/Directory1");
+    }
+
+    @Test
+    public void testChangeToParentDirectory() throws FtpException {
+        // GIVEN
+        fileSystemView.changeWorkingDirectory("/Directory1");
+
+        // WHEN
+        fileSystemView.changeWorkingDirectory("..");
+
+        // THEN
+        assertHomeDirectory("/");
+        assertCurrentDirectory("/");
+    }
+
+    @Test
+    public void testChangeToParentDirectoryNonRoot() throws FtpException {
+        // GIVEN
+        fileSystemView.changeWorkingDirectory("/Directory1");
+        fileSystemView.changeWorkingDirectory("SubDirectory1");
+
+        // WHEN
+        fileSystemView.changeWorkingDirectory("..");
+
+        // THEN
+        assertHomeDirectory("/");
+        assertCurrentDirectory("/Directory1");
+    }
+
+    @Test
+    public void testChangeToNonExistentDirectory() throws FtpException {
+        // GIVEN
+
+        // WHEN
+        boolean changeDirectoryResult = fileSystemView.changeWorkingDirectory("/Directory2/SubDirectory3/SubSubDirectory");
+
+        // THEN
+        assertFalse(changeDirectoryResult);
+        assertHomeDirectory("/");
+        assertCurrentDirectory("/");
+    }
+
+    @Test
+    public void testGetFileAbsolute() throws FtpException {
+        // GIVEN
+        fileSystemView.changeWorkingDirectory("/Directory1/SubDirectory1");
+
+        // WHEN
+        FtpFile file = fileSystemView.getFile("/Directory2/SubDirectory3");
+
+        // THEN
+        assertEquals("/Directory2/SubDirectory3", file.getAbsolutePath());
+    }
+
+    @Test
+    public void testGetFileNonAbsolute() throws FtpException {
+        // GIVEN
+        fileSystemView.changeWorkingDirectory("/Directory1/SubDirectory1");
+
+        // WHEN
+        FtpFile file = fileSystemView.getFile("SubSubDirectory");
+
+        // THEN
+        assertEquals("/Directory1/SubDirectory1/SubSubDirectory", file.getAbsolutePath());
+    }
+
+    private User createUser() {
+        BaseUser user = new BaseUser();
+        user.setName("Username");
+        user.setPassword("Password");
+        user.setHomeDirectory("/abc/def");
+        user.setAuthorities(Collections.singletonList(new WritePermission()));
+        return user;
+    }
+
+    private void assertHomeDirectory(String expectedHomeDirectory) throws FtpException {
```

Comment:
Minor: it might increase expressiveness if an Equals or EqualsWith postfix would be added. The same is truth for the one below